src/include/distributed/pg_dist_partition.h

```diff
@@ -23,6 +23,7 @@ typedef struct FormData_pg_dist_partition
 {
 	Oid logicalrelid;              /* logical relation id; references pg_class oid */
 	char partmethod;               /* partition method; see codes below */
+	int colocationid;
```

Comment:
@anarazel As far as understand you are suggesting;

- Put colocationid right after partkey and treat it as if it is variable-length field
- In LookupDistTableCacheEntry read colocationid by using heap_getattr

Am I correct? If this is the case, that would solve our problems and we will not have to recreate pg_dist_partition table.